mws/models/base.py

```diff
@@ -1,12 +1,41 @@
 """Base models for datatypes used in MWS."""
 
 from abc import ABC, abstractmethod
+from os import stat
+from typing import Any, Iterable, Mapping, Union
+from enum import Enum
+
+from mws.utils import flat_param_dict
 
 
 class MWSDataType(ABC):
     """Base class for data type models used for MWS requests."""
 
+    def __repr__(self):
+        output = f"<{self.__class__.__name__}("
+        values = [f"{key}={repr(val)}" for key, val in self.__dict__.items()]
+        output += ", ".join(values)
+        output += ")>"
+        return output
+
     @abstractmethod
-    def to_dict(self) -> dict:
-        """Returns a flattened dict of parameters suitable for an MWS request."""
+    def params_dict(self) -> dict:
+        """Returns a dict of this model's parameters suitable for an MWS request."""
         pass
+
+    def to_params(self, prefix: str = "") -> dict:
+        params_dict = self.params_dict()
+        return flat_param_dict(params_dict, prefix=prefix)
```

Comment:
Previously, `to_dict` was the only method being used, and data model had to implement it (as an abstract method). There was no machinery for including the prefix, either.

This change requires data models to implement their own `params_dict` method, which is a simpler naming convention overall. And the base class takes care of the output interface with `to_params`, which does not need to be overridden in subclasses: instead, the request methods rely on this, and can feed it a prefix as needed.

This creates a better API for this set of classes, as can be seen in how the other data models are being implemented and then used in requests.